src/raft.rs

```diff
@@ -1691,12 +1691,19 @@ impl<T: Storage> Raft<T> {
                             }
                         }
                     }
-                } else {
+                } else if m.get_from() == INVALID_ID || m.get_from() == self.id {
                     let rs = ReadState {
                         index: self.raft_log.committed,
                         request_ctx: m.take_entries()[0].take_data(),
                     };
                     self.read_states.push(rs);
+                } else {
+                    let mut to_send = Message::default();
+                    to_send.set_to(m.get_from());
+                    to_send.set_msg_type(MessageType::MsgReadIndexResp);
+                    to_send.set_index(self.raft_log.committed);
+                    to_send.set_entries(m.take_entries());
```

Comment:
Try:

```rust
to_send.to = m.from;
to_send.set_msg_type(MessageType::MsgReadIndexResp);
to_send.index = self.raft_log.committed;
to_send.entries = m.take_entries();
```

Reply:
For the style, I agree with @ice1000 , it looks better.
But I think we should unify the style later, after https://github.com/tikv/protobuf-build/pull/7 is merged and some discussion about those options in `raft-rs`.

Reply:
That's reasonable too. This comment is more like a "FYI you can do this" comment.

Reply:
ok, let's do it in another PR.